### PR TITLE
Add more ATen XLA comparison methods

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/elementwise.cpp [hunk 1]
```diff
@@ -33,16 +33,42 @@ xla::XlaOp BuildComparisonOp(const torch::jit::Node* node,
...
+    case at::aten::lt: {
+      pred = xla::Lt(input, other);
       break;
     }
     default:
-      XLA_ERROR() << "Invalid binary operator kind: " << node->kind();
+      XLA_ERROR() << "Invalid comparison operator kind: "
+                  << kind.toQualString();
   }
-  return xla::ConvertElementType(pred, xla::PrimitiveType::S8);
```
Comment (dlibenzi) [inline torch_xla/csrc/elementwise.cpp]: We should not be converting here. PRED should be coming off here.

File: torch_xla/csrc/ops/ops.cpp [hunk 2]
```diff
@@ -295,6 +295,32 @@ NodePtr AdaptiveAvgPool2dBackward(const Value& grad_output,
...
 
+NodePtr ComparisonOp(c10::Symbol kind, const Value& input, const Value& other) {
+  auto lower_fn = [kind](const ir::Node& node,
+                         ir::LoweringContext* loctx) -> ir::XlaOpVector {
+    xla::XlaOp xla_input = loctx->GetOutputOp(node.operand(0));
+    xla::XlaOp xla_other = loctx->GetOutputOp(node.operand(1));
+    xla::XlaOp xla_output = BuildComparisonOp(kind, xla_input, xla_other);
+    return node.ReturnOp(xla_output, loctx);
+  };
+  auto lower_for_shape_fn =
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/ops.cpp]: We do not need this.
The shape for a compare is the same as input.

File: torch_xla/csrc/tensor.cpp [hunk 3]
```diff
@@ -125,8 +125,9 @@ XLATensor XLATensor::Create(
   return xtensor;
 }
 
-XLATensor XLATensor::Create(ir::Value ir_value, const Device& device) {
-  XLATensor xtensor(std::move(ir_value), device);
+XLATensor XLATensor::Create(ir::Value ir_value, const Device& device,
+                            xla::Shape logical_shape) {
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.cpp]: Just take the at::ScalarType

File: torch_xla/csrc/tensor.h [hunk 4]
```diff
@@ -426,6 +463,9 @@ class XLATensor {
   static xla::int64 GetNextTensorId();
 
   std::shared_ptr<Data> data_;
+  // For some types (U8, S8 etc), the logical type of the tensor doesn't match
+  // the type of the underlying data.
+  xla::Shape logical_shape_;
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.h]: There is no need for a full shape.
This should be at::ScalarType and should be within the Data structure above.

File: torch_xla/csrc/tensor_util.cpp [hunk 5]
```diff
@@ -337,8 +338,18 @@ at::Tensor MakeTensorFromXlaLiteral(const xla::Literal& literal) {
           literal, at::ScalarType::Short, xla::PrimitiveType::S16);
     }
     case xla::PrimitiveType::S32: {
-      return XlaLiteralToTensor<xla::int32, int32_t>(
-          literal, at::ScalarType::Int, xla::PrimitiveType::S32);
+      switch (dest_element_type) {
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_util.cpp]: Given that we support tensor rewriting at this point, this should be fully (or almost fully) populated.

File: torch_xla/csrc/tensor.h (hunk 4 above)
Comment (asuhan) [reply torch_xla/csrc/tensor.h]: What would `shape()` do then, re-create a shape with the changed element type every time we call it? That might be costly.

File: torch_xla/csrc/ops/ops.cpp (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/ops/ops.cpp]: With the element type changed to `PRED` though, which is why I did it this way. But I can do it the other way around.

File: torch_xla/csrc/tensor.cpp (hunk 3 above)
Comment (asuhan) [reply torch_xla/csrc/tensor.cpp]: See comment below. I was inclined to only store the logical element type, but then `XLATensor::shape()` becomes a problem.

File: torch_xla/csrc/tensor_util.cpp (hunk 5 above)
Comment (asuhan) [reply torch_xla/csrc/tensor_util.cpp]: What do you mean?

File: torch_xla/csrc/tensor.cpp (hunk 3 above)
Comment (asuhan) [reply torch_xla/csrc/tensor.cpp]: Done.

File: torch_xla/csrc/ops/ops.cpp (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/ops/ops.cpp]: Done.

File: torch_xla/csrc/elementwise.cpp (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/elementwise.cpp]: Done.

File: torch_xla/csrc/tensor.h (hunk 4 above)
Comment (asuhan) [reply torch_xla/csrc/tensor.h]: Done.

**Pull request merged**